### PR TITLE
Default-enable BalsaParser.

### PR DESCRIPTION
**Pull request by bencebeky**
This feature has been widely deployed by two large projects at different companies without any production issues. A third project has enabled it in staging and in end-to-end tests. Manual load testing does not show significant change in CPU or memory usage compared to the legacy HTTP/1 parser.

Tracking issue: #21245

Commit Message: Default-enable BalsaParser.
Additional Description:
Risk Level: medium
Testing: unit tests and e2e test within Envoy and by multiple embedders
Docs Changes: n/a
Release Notes: Default-enable BalsaParser.
Platform Specific Features: n/a

### CODE REVIEW

**Comment by bencebeky** [comment]
> Does this enable balsa for server or client codecs, or both?

Both.

**Comment by yanavlasov** [comment]
Would you like to wait until we get all clear from the differential fuzzer? I have just assigned you another breaking change bug internally, so it seems like there are still behaviors in the new codec that may cause significant compatibility issues.

File: changelogs/current.yaml [hunk 1]
```diff
@@ -14,6 +14,9 @@ behavior_changes:
     :ref:`initial_fetch_timeout <envoy_v3_api_field_config.core.v3.ConfigSource.initial_fetch_timeout>` times out, and will then apply
     the cached assignment and finish updating the warmed cluster. This change is disabled by default, and can be enabled by setting
     the runtime flag ``envoy.restart_features.use_eds_cache_for_ads`` to true.
+- area: http
+  change: |
+    Flip default value of ``envoy.reloadable_features.http1_use_balsa_parser`` runtime flag from false to true.
```
Comment (alyssawilk) [inline changelogs/current.yaml:19]: I think we want this more informative.
"Switching from http_parser to balsa_parser for handling HTTP/1.1 traffic.  See [issue url] for details.  this behavioral change can be reverted by setting  ``envoy.reloadable_features.http1_use_balsa_parser``  to false.

**Review by alyssawilk — COMMENTED** [review]
wohoo!
/wait

**Comment by alyssawilk** [comment]
@ggreenway can we get a !google stamp on the flip?
Motivation is in the linked issue. We've closed all open fuzz bugs so are very happy with the lack of behavioral differences, plus canaried within Google and pinterest

**Review by alyssawilk — APPROVED** [review]
!!

**Review by ggreenway — APPROVED** [review]
Awesome! Glad to be moving off of abandonware.

**Pull request merged**